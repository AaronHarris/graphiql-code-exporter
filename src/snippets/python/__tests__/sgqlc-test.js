import snippetObject from '../sgqlc';

import getOptionCombinations from '../../__helpers__/getOptionCombinations';

const {options, generate} = snippetObject;
const optionCombinations = getOptionCombinations(options);

const testQuery = `
query testQuery {
  someData {
    id
  }
}
`;

const testMutation = `
mutation testMutation {
  addData(id: "id") {
    id
  }
}`;

describe('Generating a Python:react-apollo snippet', () => {
  it('should generate the correct query snippet', () => {
    optionCombinations.forEach(options => {
      const snippet = generate({
        serverUrl: 'https://api.myservice.com/',
        operation: testQuery,
        operationType: 'query',
        variableName: 'TEST_QUERY',
        operationName: 'testQuery',
        options,
      });

      expect({
        options,
        snippet: '\n' + snippet,
      }).toMatchSnapshot();
    });
  });

  it('should generate the correct mutation snippet', () => {
    optionCombinations.forEach(options => {
      const snippet = generate({
        serverUrl: 'https://api.myservice.com/',
        operation: testMutation,
        operationType: 'mutation',
        variableName: 'TEST_MUTATION',
        operationName: 'testMutation',
        options,
      });

      expect({
        options,
        snippet: '\n' + snippet,
      }).toMatchSnapshot();
    });
  });
});
