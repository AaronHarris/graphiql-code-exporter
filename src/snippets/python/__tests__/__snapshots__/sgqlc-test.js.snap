// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generating a Python:react-apollo snippet should generate the correct mutation snippet 1`] = `
Object {
  "options": Object {},
  "snippet": "

from sgqlc.endpoint.http import HTTPEndpoint

serverUrl = 'https://api.myservice.com/'

query = '''

mutation testMutation {
  addData(id: \\"id\\") {
    id
  }
}
'''

endpoint = HTTPEndpoint(serverUrl, undefined)
data = endpoint(query, undefined)

print(data)
",
}
`;

exports[`Generating a Python:react-apollo snippet should generate the correct query snippet 1`] = `
Object {
  "options": Object {},
  "snippet": "

from sgqlc.endpoint.http import HTTPEndpoint

serverUrl = 'https://api.myservice.com/'

query = '''

query testQuery {
  someData {
    id
  }
}

'''

endpoint = HTTPEndpoint(serverUrl, undefined)
data = endpoint(query, undefined)

print(data)
",
}
`;
